addon/MetadataStore.js

```diff
@@ -525,6 +525,38 @@ MetadataStore.prototype = {
   }),
 
   /**
+   * Find the oldest entry in the database
+   *
+   * Returns the timestamp of the oldest entry in the database
+   */
+  asyncGetOldestInsert: Task.async(function*() {
+    let timestamp = null;
+    try {
+      const entry = yield this.asyncExecuteQuery("SELECT min(created_at) FROM page_metadata");
+      if (entry && entry.length) {
+        timestamp = entry[0][0];
+      }
+    } catch (e) {
+      throw e;
+    }
+    return timestamp;
+  }),
+
+  /**
+   * Counts all the items in the database
+   *
+   * Returns a promise with the array of the retrieved metadata records
+   */
+  asyncCountAllItems: Task.async(function*() {
+    try {
+      const result = yield this.asyncExecuteQuery("SELECT count(*) FROM page_metadata");
+      return result;
```

Comment:
We have to unpack this array of arrays here:

 {"source":"PLACES_STATS_FEED","event_id":"{a7e1b003-00d4-f44e-91cd-46459da20e8a}","action":"activity_stream_performance","tab_id":"-5-1","event":"countMetadataURLs","value":[[310]],"client_id":"c3a2a51b-91cf-0243-b882-03763cf9761d","addon_version":"1.1.8","locale":"en-US","page":"NEW_TAB"}